examples/simulate_fight_scenario.py

```diff
@@ -0,0 +1,111 @@
+            self.control_requested = True
+
+        if self.control_requested and self.enemy_units and not self.control_received:
+            # prepare my side
+            me = 1
+            cc = self.townhalls.first
+            p = cc.position.towards(self.game_info.map_center, 3)
+            # create supply
+            await self._client.debug_create_unit([[UnitTypeId.SUPPLYDEPOT, 1, p, me]])
+            # destroy command center
+            await self._client.debug_kill_unit([cc.tag])
+            # destroy all workers
+            for w in self.units(UnitTypeId.SCV):
+                await self._client.debug_kill_unit([w.tag])
+            # create marines
+            await self._client.debug_create_unit([[UnitTypeId.MARINE, 4, p, me]])
+
+            # prepare opponent side
+            pc = 2
+            cc = self.enemy_structures.first
+            p = cc.position.towards(self.game_info.map_center, 3)
+            # create supply
+            await self._client.debug_create_unit([[UnitTypeId.SUPPLYDEPOT, 1, p, pc]])
+            # destroy command center
+            await self._client.debug_kill_unit([cc.tag])
+            # destroy all workers
+            for w in self.enemy_units(UnitTypeId.SCV):
+                await self._client.debug_kill_unit([w.tag])
+            # create marines
+            await self._client.debug_create_unit([[UnitTypeId.MARINE, 4, p, pc]])
+            logger.info("control received")
+            # await self.chat_send("control received")
+            self.control_received = True
+
+        # to speedup, we are going damage both supplies
+        if not self.supplies_been_damaged and self.structures(UnitTypeId.SUPPLYDEPOT) and self.enemy_structures(UnitTypeId.SUPPLYDEPOT):
+            for s in self.structures:
+                await self._client.debug_set_unit_value([s.tag], 2, 100)
+            for s in self.enemy_structures:
+                await self._client.debug_set_unit_value([s.tag], 2, 100)
+            logger.info("supplies damaged")
+            # await self.chat_send("supplies damaged")
+            self.supplies_been_damaged = True
+
+        # note: we should wait till workers will be destroyed
+        if not self.fight_started and self.control_received and self.enemy_units and not self.enemy_units(UnitTypeId.SCV) and not self.units(UnitTypeId.SCV):
+            # start fight
+            for u in self.enemy_units:
+                u.attack(self.structures.first.position)
+            for u in self.units:
+                u.attack(self.enemy_structures.first.position)
+            # await self._client.move_camera(self._game_info.map_center)
+            logger.info("fight started")
+            # await self.chat_send("fight started")
+            self.fight_started = True
+
+        for u in self.units(UnitTypeId.MARINE):
+            u.attack(self.enemy_structures.first.position)
+            # TODO: implement your fight logic here
```

Comment:
Would perhaps be better to have 2 bots fight each other where they can have different micro actions to be able to see more easily which micro is "better".

Reply:
Depends on end goal, in my case I want to beat computer on hard level, to check if everything is ok at the end I will play against him

> Idea of having to bots compeat each other is good for some kind of evolution and compassion and is something on a next level for me to be fair